google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -324,7 +329,7 @@ LeakedConnectionException getLeakedException() {
 
   @Override
   public Dialect getDialect() {
-    return options.getDialect();
+    return dbClient.getDialect();
```

Comment:
I remember reading that the getDialect will be supported through options. Just putting a note in case my understanding is not correct.

Reply:
The options will not support a `getDialect` method, but there is a `SessionPoolOptions#setAutoDetectDialect(boolean)`. Setting that option to true will make sure that the dialect detection query is executed automatically at startup in the background. That will normally make the `DatabaseClient#getDialect` method non-blocking, unless it is called directly after creating a database client and the execution of the dialect query is still going on in the background. So to summarize:
1. `getDialect()` can always be called.
2. The result of `getDialect` will always be cached.
3. The first call to `getDialect` can be blocking, unless `SessionPoolOptions#setAutoDetectDialect(true)` has been called.